lib/beautiful/gtk.lua

```diff
@@ -0,0 +1,169 @@
+      bg_color="background-color",
+      fg_color="color",
+    }
+  ))
+  result = join(result, read_gtk_color_properties_from_widget(
+    Gtk.Entry(), {
+      base_color="background-color",
+      text_color="color",
+    }
+  ))
+  result = join(result, read_gtk_color_properties_from_widget(
+    Gtk.ToggleButton(), {
+      selected_bg_color="background-color",
+      selected_fg_color="color",
+    }
+  ))
+  local button = Gtk.Button()
+  result = join(result, read_gtk_color_properties_from_widget(
+    button, {
+      button_bg_color="background-color",
+      button_fg_color="color",
+    }
+  ))
+  local button_style_context = button:get_style_context()
+  for result_key, style_context_property in pairs({
+    border_radius="border-radius",
+    border_width="border-top-width",
+  }) do
+    local state = style_context:get_state()
+    local property = button_style_context:get_property(style_context_property, state)
+    result[result_key] = property.value
+    property:unset()
+  end
+
+  for _, color_data in ipairs({
+    {"bg_color", "theme_bg_color"},
+    {"fg_color", "theme_fg_color"},
+    {"base_color", "theme_base_color"},
+    {"text_color", "theme_text_color"},
+    {"selected_bg_color", "theme_selected_bg_color"},
+    {"selected_fg_color", "theme_selected_fg_color"},
+    --
+    {"tooltip_bg_color", "theme_tooltip_bg_color", "bg_color"},
+    {"tooltip_fg_color", "theme_tooltip_fg_color", "fg_color"},
+    {"osd_bg_color", "osd_bg", "tooltip_bg_color"},
+    {"osd_fg_color", "osd_fg", "tooltip_fg_color"},
+    {"osd_border_color", "osd_borders_color", "osd_fg_color"},
+    {"menubar_bg_color", "menubar_bg_color", "bg_color"},
+    {"menubar_fg_color", "menubar_fg_color", "fg_color"},
+    --
+    {"button_bg_color", "button_bg_color", "bg_color"},
+    {"button_fg_color", "button_fg_color", "fg_color"},
+    {"header_button_bg_color", "header_button_bg_color", "menubar_bg_color"},
+    {"header_button_fg_color", "header_button_fg_color", "menubar_fg_color"},
+    --
+    {"wm_bg_color", "wm_bg", "menubar_bg_color"},
+    {"wm_border_focused_color", "wm_border_focused", "selected_bg_color"},
+    {"wm_title_focused_color", "wm_title_focused", "menubar_bg_color"},
+    {"wm_icons_focused_color", "wm_icons_focused", "menubar_fg_color"},
+    {"wm_border_unfocused_color", "wm_border_unfocused", "wm_bg_color"},
+    {"wm_title_unfocused_color", "wm_title_unfocused", "menubar_bg_color"},
+    {"wm_icons_unfocused_color", "wm_icons_unfocused", "menubar_fg_color"},
+  }) do
+    local result_key, style_context_key, fallback_key = unpack(color_data)
+    result[result_key] = lookup_gtk_color_to_hex(style_context, style_context_key) or
+      (result[result_key] ~= "#000000" and result[result_key] or result[fallback_key] or result[result_key])
+    if not result[result_key] then
+        gears_debug.print_warning("Can't read color '" .. style_context_key .. "' from GTK+3 theme.")
+    end
+  end
+
+  window:destroy()
```

Comment:
Is this `:destroy()` necessary? If not, I'd drop it.

Reply:
i think it does the same as closing the window (if it would be visible)